components/tabs/android/src/main/java/mozilla/appservices/remotetabs/RemoteTabsProvider.kt

```diff
@@ -0,0 +1,152 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.appservices.remotetabs
+
+import com.sun.jna.Native
+import com.sun.jna.Pointer
+import mozilla.appservices.remotetabs.rust.LibRemoteTabsFFI
+import mozilla.appservices.support.native.toNioDirectBuffer
+import mozilla.appservices.remotetabs.rust.RustError
+import mozilla.appservices.sync15.SyncTelemetryPing
+import java.util.concurrent.atomic.AtomicLong
+
+class RemoteTabsProvider(localDeviceId: String) : AutoCloseable {
+    private var handle: AtomicLong = AtomicLong(0)
+
+    init {
+        handle.set(rustCall { error ->
+            LibRemoteTabsFFI.INSTANCE.remote_tabs_new(localDeviceId, error)
+        })
+    }
+
+    /**
+     * Update our local tabs state.
+     */
+    fun updateLocalState(state: List<RemoteTab>) {
```

Comment:
What do you think about calling this `setLocalTabs` or `sendLocalTabs` (`pushLocalTabs`?), or something like that? `updateLocalState` almost implies we're updating the synced tab list with remote tabs, when it's the other way around.